### PR TITLE
[otbn] Move, document and tidy up otbn-rig

### PR DESCRIPTION
**Pull request by rswarbrick**
This probably belongs in the `dv` directory, so move it there. Also, update the `README.md` (that I had completely forgotten about) and remove an unnecessary argument.

### CODE REVIEW

File: hw/ip/otbn/dv/rig/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+loops. The JSON output format represents this tree of snippets.
+
+### Initialised data
+
+The generated program is designed never to trigger architecturally
+unspecified behaviour. It might trigger errors, but its execution will
+never depend on things like the initial contents of the register file
+or uninitialised memories.
+
+If there is no initialised data in the image, this means any load
```
Comment (imphil) [inline hw/ip/otbn/dv/rig/README.md:68]:
```suggestion
If there would be no initialised data in the image, this means any load
```

Or swap the paragraphs 2 and 3 to have the implemented functionality first, and the motivation after that.

File: hw/ip/otbn/dv/rig/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+To work around this, the generated program also includes a few words
+of (randomly) initialised data, scattered around dmem.
+
+## The asm command
+
+Once a random program has been generated, the `asm` command can be
+used to convert it to an assembly listing. This, in turn, can be
+assembled and linked using the toolchain in `hw/ip/otbn/util`.
+
+Unlike the `gen` command, this step is completely deterministic.
```
Comment (imphil) [inline hw/ip/otbn/dv/rig/README.md:85]: Isn't `gen` deterministic as long as the same seed is given?

**Review by imphil — APPROVED** [review]
LGTM, thanks!

File: hw/ip/otbn/dv/rig/README.md (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/otbn/dv/rig/README.md]: Yeah, I rewrote this paragraph three times trying to come up with a nice way to say things. I've just pushed up a new version that gives up and doesn't provide as much detail. Much easier to read, and anyone who wants to know more can read the comments in the code!

File: hw/ip/otbn/dv/rig/README.md (hunk 2 above)
Comment (rswarbrick) [reply hw/ip/otbn/dv/rig/README.md]: Good point! I've rewritten things to make it a bit more sensible.

**Pull request merged**